src/main/java/org/jenkinsci/plugins/durabletask/executors/ContinuedTask.java

```diff
@@ -60,6 +76,10 @@ private static boolean isContinued(Queue.Task task) {
             if (isContinued(item.task)) {
                 return null;
             }
+            Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null || jenkins.getInitLevel().compareTo(InitMilestone.COMPLETED) < 0) {
```

Comment:
So when the queue is [instantiated the MaintainTask is scheduled](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/hudson/model/Queue.java#L308)... that's going to [run every 5 seconds](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/hudson/model/Queue.java#L2706)

The Queue is instantiated [when Jenkins is being constructed](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/jenkins/model/Jenkins.java#L774) after that [we execute the reactor](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/jenkins/model/Jenkins.java#L816) and some time later we [update the computer list](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/jenkins/model/Jenkins.java#L844) which will also [schedule a maintenance](https://github.com/jenkinsci/jenkins/blob/ec95aa5b21ab8d5bc0a74bf7553b57523e76766f/core/src/main/java/hudson/model/AbstractCIBase.java#L207)

The key point here is that the 5 second time starts counting down once the Queue is instantiated... so if the reactor takes more than 5 seconds to get to `COMPLETED` then the queue will start doing maintenance before the reactor is complete...

Sounds like a possible bug... for sure I think it would be better to have the maintenance skipped until we are completed... but until the baseline version of Jenkins has such a fix I believe this is necessary